pkg/reloader/reloader.go

```diff
@@ -85,6 +85,7 @@ import (
 type Reloader struct {
 	logger        log.Logger
 	reloadURL     *url.URL
+	HTTPClient    http.Client
```

Comment:
Is private field sufficient here?

Reply:
Didnt get you ? :thinking:

Reply:
Do we need to expose `HTTPClient` field? If not, we can rename it to `httpClient`

Reply:
Yeah, I didn't wanna make it public field either but had assign it from outside once we are in `runSidecar` func in sidecar.go so that's there, any better approach here ? Configuring http client beforehand and passing it to reloader and runSidecar func ?

Reply:
Got it. Then you can have another public method to set httpclient for `Reloader`.